### DISCUSSION TITLE
Display the Neo-Tree in a position similar to netrw after executing the nvim . command.

### DISCUSSION THREAD

**Discussion by MatejBransky** [LazyVim/LazyVim#2366] [Q&A] [answered]
Currently, when you enter nvim . in a directory, it loads an empty buffer with the NeoTree explorer displayed on the left side. However, I would prefer it to open in the 'position = "current"'. Is that possible?

Thank you for any help. 🙏

**Comment by dpetka2001** [comment, marked as answer]
```lua
-- ~/.config/nvim/lua/plugins/neo-tree.lua
return {
  {
    "neo-tree.nvim",
    init = function()
      if vim.fn.argc(-1) == 1 then
        local stat = vim.loop.fs_stat(vim.fn.argv(0))
        if stat and stat.type == "directory" then
          require("neo-tree").setup({
            window = {
              position = "current",
            },
          })
        end
      end
    end,
  },
}
```
Try this out if it does what you want.

**Reply by MatejBransky** [reply]
It works! Thank you so much!

Is there a way to wait for Neo-Tree to finish loading? Currently, there's a brief glitch where the old netrw is visible for a few milliseconds.

https://github.com/LazyVim/LazyVim/assets/20944757/e45913f3-ef48-4d6d-9b8a-8bf214a623c3

**Reply by dpetka2001** [reply]
Not as far as I know (maybe someone else knows of a better solution I'm not aware of). You could try disabling `netrw` builtin plugin, so you don't see it when the glitch happens, but the glitch will still momentarily happen but with an empty buffer before loading `neo-tree` if you choose to disable `netrw` after all.

**Reply by MatejBransky** [reply]
Thanks. I've found this solution and I've noticed some option related to the demanded behavior in the neo-tree config:

```lua
{
  "nvim-neo-tree/neo-tree.nvim",
  init = function()
    vim.api.nvim_create_autocmd("BufEnter", {
      -- make a group to be able to delete it later
      group = vim.api.nvim_create_augroup("NeoTreeInit", { clear = true }),
      callback = function()
        local f = vim.fn.expand("%:p")
        if vim.fn.isdirectory(f) ~= 0 then
          vim.cmd("Neotree current dir=" .. f)
          -- neo-tree is loaded now, delete the init autocmd
          vim.api.nvim_clear_autocmds({ group = "NeoTreeInit" })
        end
      end,
    })
    -- keymaps
  end,
  opts = {
    filesystem = {
      hijack_netrw_behavior = "open_current",
    }
  }
}
```

With this setup I can keep netrw enabled and the empty buffer before the neo-tree loading doesn't look so bad. 🙂 

https://github.com/LazyVim/LazyVim/assets/20944757/ff7c39fa-f68a-4724-831f-21bf8f681c27

Nevertheless thank you for you help! 🙏

**Reply by dpetka2001** [reply]
Cool!! Tnx for sharing this :smile:

**Reply by Kayzels** [reply]
I've tried this now, but I'm getting the weird thing of Neotree appearing twice: once on the left, and then once as the main panel.

I've copied over the exact code you wrote above: is there something else I need to change to make it work?
